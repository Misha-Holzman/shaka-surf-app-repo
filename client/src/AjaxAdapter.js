const axios = require('axios');

module.exports = (baseURL, noun, headers) => {
  const config = {
    baseURL,
    headers,
    timeout: 1000,
  };

  const instance = axios.create(config);

  return {
    config,
    create(data) {
      return instance.post('/', { [noun]: data })
        .catch((e) => { throw e; });
    },

    async read() {
      try {
        const { data } = await instance.get('/');
        return data[`${noun}s`];
      } catch (e) {
        throw e;
      }
    },

    update(id, data) {
      return instance.put(`/${id}`, { [noun]: data })
        .catch((e) => { throw e; });
    },

    destroy(id) {
      return instance.delete(`/${id}`)
        .catch((e) => { throw e; });
    },
  };
};
