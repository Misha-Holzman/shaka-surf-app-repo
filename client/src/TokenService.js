/* eslint-env browser */
const tokenService = {
  set token(token) {
    // store token in localStorage under the 'authToken' key
    localStorage.setItem('authToken', token);
  },
  get token() {
    // fetch the token out of localStorage
    return localStorage.getItem('authToken');
  },

  destroy() {
    // delete the token
    localStorage.removeItem('authToken');
  },

};
module.exports = tokenService;
