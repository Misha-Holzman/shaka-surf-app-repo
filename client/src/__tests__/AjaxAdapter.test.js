/* eslint-env jest */
import TokenService from '../TokenService';
import AjaxAdapter from '../AjaxAdapter';

describe('AjaxAdapter', () => {
  let TweetService;

  beforeEach(() => {
    TweetService = AjaxAdapter('/api/tweets', 'tweets', {
      "Content-Type": "application/json",
      get Authorization(){
        return `Bearer ${TokenService.token}`;
      },
    });
  });

  it('should preserve a getter after destructure', () => {
    let total = 0;
    const getterObj = {
      get test() {
        return total;
      }
    }
    total = 4;
    expect(getterObj.test).toBe(4);
  })

  it('should preserve a getter after restructure', () => {
    let total = 0;
    const getterObj = {
      get test() {
        return total;
      }
    }

    const newGetterObj = {
      getterObj,
    }
    total = 4
    expect(newGetterObj.getterObj.test).toBe(4)
  })

  it('should create an AjaxAdapter', () => {
    expect(TweetService).toMatchSnapshot();
  });

  it('should contain a getter in the Authorization header', () => {
    const myToken = 'foo:bar';
    TokenService.token = myToken;

    debugger;
    //const spy = jest.spyOn(TweetService.config.headers, 'Authorization', 'get')
    expect(TweetService.config.headers.Authorization).toMatch(myToken);
  });
});
