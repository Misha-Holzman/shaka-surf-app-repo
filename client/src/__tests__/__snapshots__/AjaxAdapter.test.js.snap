// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AjaxAdapter should call the constructor 1`] = `
Array [
  Object {
    "baseURL": "/api/tweets",
    "headers": Object {
      "Authorization": "Bearer my-dummy-token",
      "Content-Type": "application/json",
    },
    "timeout": 1000,
  },
]
`;

exports[`AjaxAdapter should create an AjaxAdapter 1`] = `
Object {
  "config": Object {
    "baseURL": "/api/tweets",
    "headers": Object {
      "Authorization": "Bearer my-dummy-token",
      "Content-Type": "application/json",
    },
    "timeout": 1000,
  },
  "create": [Function],
  "destroy": [Function],
  "read": [Function],
  "update": [Function],
}
`;

exports[`AjaxAdapter should remove the token in header when no token is present 1`] = `
Object {
  "Authorization": "Bearer null",
  "Content-Type": "application/json",
}
`;
